src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -76,13 +76,13 @@ internal interface IDiagnosticAnalyzerService
         /// This API will only force complete analyzers that support span based analysis, i.e. compiler analyzer and
         /// <see cref="IBuiltInAnalyzer"/>s that support <see cref="DiagnosticAnalyzerCategory.SemanticSpanAnalysis"/>.
         /// For the rest of the analyzers, it will only return diagnostics if the analyzer has already been executed.
-        /// Use <see cref="GetDiagnosticsForSpanAsync(TextDocument, TextSpan?, Func{string, bool}?, bool, bool, CodeActionRequestPriority, Func{string, IDisposable?}?, DiagnosticKind, bool, CancellationToken)"/>
+        /// Use <see cref="GetDiagnosticsForSpanAsync(TextDocument, TextSpan?, Func{string, bool}?, bool, bool, CodeActionRequestPriorityProvider?, Func{string, IDisposable?}?, DiagnosticKind, bool, CancellationToken)"/>
         /// if you want to force complete all analyzers and get up-to-date diagnostics for all analyzers for the given span.
         /// </summary>
         Task<(ImmutableArray<DiagnosticData> diagnostics, bool upToDate)> TryGetDiagnosticsForSpanAsync(
             TextDocument document, TextSpan range, Func<string, bool>? shouldIncludeDiagnostic,
             bool includeSuppressedDiagnostics = false,
-            CodeActionRequestPriority priority = CodeActionRequestPriority.None,
+            CodeActionRequestPriorityProvider? priorityProvider = null,
```

Comment:
Note: #67434 tracks making all parameters for methods on this interface non-optional.